### DISCUSSION TITLE
GitHub Action for setting up detekt

### DISCUSSION THREAD

**Discussion by peter-murray** [detekt/detekt#3369] [Show and tell]
FYI, I have built a second GitHub Action that utilizes the native JavaScript frameworks for GitHub Actions and tool caching aspects of GitHub Actions to provide detekt to a GitHub Action runner.

https://github.com/marketplace/actions/setup-detekt

It is more in line with the other `setup-xxx` GitHub Actions and will be faster to use in GitHub Actions workflows compared to the Docker based action that exists today.

**Comment by picklebento** [comment]
Thank you for publishing an optimized Github Action. Once the next release version is published, we should be able to upload SARIF to Github code scanning (https://github.com/detekt/detekt/pull/3359)

**Comment by cortinico** [comment]
That's a great addition @peter-murray. Do you want to update the README to link your custom Action alongside the other?

**Reply by peter-murray** [reply]
That would be great if you could 😄

**Reply by cortinico** [reply]
Added here https://github.com/detekt/detekt/pull/3373
